boot/freeldr/freeldr/ntldr/wlmemory.c

```diff
@@ -238,6 +239,65 @@ WinLdrSetupMemoryLayout(IN OUT PLOADER_PARAMETER_BLOCK LoaderBlock)
         return FALSE;
     }
 
+    // Always map the ACPI table if present, otherwise Windows will crash.
+    if (Extension->AcpiTable)
+    {
+        PVOID AcpiTableClone = MmAllocateMemoryWithType(Extension->AcpiTableSize, LoaderFirmwarePermanent);
+        if (!AcpiTableClone)
+        {
+            ERR("Cannot allocate ACPI table\n");
+            return FALSE;
+        }
+
+        RtlCopyMemory(AcpiTableClone, Extension->AcpiTable, Extension->AcpiTableSize);
+        Extension->AcpiTable = AcpiTableClone;
```

Comment:
I wonder whether it could be possible to just do:
```suggestion
        MmSetMemoryType(Extension->AcpiTable, Extension->AcpiTableSize, LoaderFirmwarePermanent);
```
and doing something similar as well for each entries below.

Reply:
Do you think that will work because the table is the very end of the memory and freeldr cannot map this area.

Reply:
It will say that table is out of memory.

Reply:
This looks like a bug or weird limitation of freeldr itself.
Since other memory can correctly be marked.